src/OrchardCore/Orchard.ContentManagement/ContentFieldFactory.cs

```diff
@@ -1,25 +1,35 @@
 ﻿using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
-using System.Linq;
-using Orchard.ContentManagement;
 
-public class ContentFieldFactory : IContentFieldFactory
+namespace Orchard.ContentManagement
 {
-    private readonly IEnumerable<ContentField> _contentFields;
-    
-    public ContentFieldFactory(IEnumerable<ContentField> contentFields)
+    /// <summary>
+    /// Implements <see cref="ITypeActivatorFactory{ContentField}"/> by resolving all registered <see cref="ContentField"/> types
+    /// and memoizing a statically typed <see cref="ITypeActivator{ContentField}"/>.
+    /// </summary>
+    public class ContentFieldFactory : ITypeActivatorFactory<ContentField>
     {
-        _contentFields = contentFields;
-    }
+        private ITypeActivator<ContentField> ContentFieldActivator = new GenericTypeActivator<ContentField, ContentField>();
 
-    public ContentField CreateContentField(string FieldName)
-    {
-        var contentField = _contentFields.FirstOrDefault(x => x.GetType().Name == FieldName);
-        if (contentField != null)
-        { 
-            return (ContentField)Activator.CreateInstance(contentField.GetType());
+        private readonly ConcurrentDictionary<string, ITypeActivator<ContentField>> _contentFieldActivators;
```

Comment:
Actually move this comment to the ContentPart one, but you are right for fields, not needed